File: apps/home-service-platform-node-js-admin/src/api/booking/BookingCreateInput.ts
import { AdminCreateNestedManyWithoutBookingsInput } from "./AdminCreateNestedManyWithoutBookingsInput";
import { ServiceCategoryWhereUniqueInput } from "../serviceCategory/ServiceCategoryWhereUniqueInput";
import { ServiceProviderWhereUniqueInput } from "../serviceProvider/ServiceProviderWhereUniqueInput";
import { ServiceProviderCreateNestedManyWithoutBookingsInput } from "./ServiceProviderCreateNestedManyWithoutBookingsInput";
import { UserWhereUniqueInput } from "../user/UserWhereUniqueInput";

export type BookingCreateInput = {
  admins?: AdminCreateNestedManyWithoutBookingsInput;
  bookingDate?: Date | null;
  paymentStatus?: string | null;
  serviceCategory?: ServiceCategoryWhereUniqueInput | null;
  serviceProvider?: ServiceProviderWhereUniqueInput | null;
  serviceProviders?: ServiceProviderCreateNestedManyWithoutBookingsInput;
  status?: string | null;
  user?: UserWhereUniqueInput | null;
};

File: apps/home-service-platform-server/src/APIs/Booking/Dtos/BookingCreateInput.cs
namespace HomeServicePlatform.APIs.Dtos;

public class BookingCreateInput
{
    public DateTime? BookingDate { get; set; }

    public DateTime CreatedAt { get; set; }

    public string? Id { get; set; }

    public string? PaymentStatus { get; set; }

    public ServiceCategory? ServiceCategory { get; set; }

    public ServiceProvider? ServiceProvider { get; set; }

    public List<ServiceProvider>? ServiceProviders { get; set; }

    public string? Status { get; set; }

    public DateTime UpdatedAt { get; set; }

    public User? User { get; set; }

    public List<User>? Users { get; set; }
}
